### ISSUE TITLE
Rename `latest`

### ISSUE THREAD

**Issue by astrojuanlu** [readthedocs/readthedocs.org#10674]
## What's the problem this feature will solve?
We've had problems in the past with users thinking that `latest` meant "latest released version" when in fact means "latest development version" (https://github.com/kedro-org/kedro/issues/2823).

## Describe the solution you'd like
We'd like to rename `latest` to something else, for example `unreleased` (very clear), `development` (not as good), `nightly` (still a term used mostly by programmers, not ideal) or similar.

## Alternative solutions
@humitos already proposed some things we can do https://github.com/kedro-org/kedro/issues/2823#issuecomment-1642583278 but I think none of this fulfills what we want:

> * make `latest` to point to the branch/tag you want from the project's setting by defining the "Default branch"

This repurposes `latest` to do something else.

> * disable `latest` version completely if you want

We do use `latest`, so we don't want to disable it. We want to change its name.

> * set "Default version" to something different than `latest` (this is the "Default version" setting)

Our "Default version" is already `stable`, our problem is not there.

> * override `latest` by creating a branch/tag with that name

This still doesn't cut it.

**Comment by humitos** [comment]
If you disable `latest` and make the "Default version" to point `main` -- does that resolves the confusion?

**Comment by humitos** [comment]
You can also create a `development` (or whatever that communicates that's the "development branch") branch in your project and make "Default version" to point to `development`. After that, `latest` can be disabled.

**Comment by humitos** [comment]
> We'd like to rename `latest` to something else

I'm sure this won't be possible in the near future since its name it's pretty hardcoded in multiple places of the code. _However_, we have been working a lot on El Proxito (the code that serves the documentation) and we are making progress on being able to configure how the URLs look like. So, eventually, it may be possible to "rename a version", but I don't think this will happen _soon_.

Also, this issue is kind of related of "having an alias" for a version, which has a similar complexity: https://github.com/readthedocs/readthedocs.org/pull/6273

**Comment by astrojuanlu** [comment]
> If you disable latest and make the "Default version" to point to main -- does that resolves the confusion?

Not really, we'd still want the default version to be `stable`, hence the true "latest" released version.

> You can also create a development (or whatever that communicates that's the "development branch") branch in your project and make "Default version" to point to development. After that, latest can be disabled. Would this be useful for this use case?

Similarly, we'd want our "Default version" to be `stable`.

Our development branch is called `main` like virtually any other GitHub project out there, so enabling the `main` version still wouldn't help a lot.

As a workaround we could have a `unreleased` branch be in sync with `main` and then enable it on RTD, but it's kind of an ugly hack.

> Also, this issue is kind of related of "having an alias" for a version, which has a similar complexity: https://github.com/readthedocs/readthedocs.org/pull/6273

Yeah changing a version slug (what https://github.com/readthedocs/readthedocs.org/pull/4505 started) is exactly what would solve this issue.

> I'm sure this won't be possible in the near future since its name it's pretty hardcoded in multiple places of the code.

Understood - feel free to keep this issue open or close it, as you prefer. Thanks @humitos!

**Comment by humitos** [comment]
> Not really, we'd still want the default version to be `stable`, hence the true "latest" released version.

Ah, OK. Forget about changing the "Default version". I thought you were using `latest` as "Default version". Never mind what I said about changing the "Default version" at all 😄 

> As a workaround we could have a `unreleased` branch be in sync with `main` and then enable it on RTD, but it's kind of an ugly hack.

Unfortunately, I think this is the only way to achieve your goal currently.

**Comment by astrojuanlu** [comment]
Yay, thanks @stsewd and team! Could you enable it for `kedro` for now? I think this should be enough for our purposes

**Comment by stsewd** [comment]
Done!

**Comment by astrojuanlu** [comment]
I just set the slug of `main` to `nightly` 💯 maybe 1 thing that would make things easier is to see it in the UI:

![Image](https://github.com/user-attachments/assets/5abf900f-548e-4ac6-b372-a4576a201e35)

Like `main (nightly)` or sth like that.

And another thing, we see the "might be an old version of the docs" banner

![Image](https://github.com/user-attachments/assets/b88e5a97-5018-4eeb-9f8d-56f92c8d1d71)

but I'd actually love to have a bit of control over that (happy to open new issues if needed)

**Comment by stsewd** [comment]
maybe @agjohnson has some thoughts about how to distinct that in the UI.

> And another thing, we see the "might be an old version of the docs" banner
> but I'd actually love to have a bit of control over that (happy to open new issues if needed)

Maybe @humitos has a better answer for that.

**Comment by humitos** [comment]
> And another thing, we see the "might be an old version of the docs" banner

Yeah, the logic for the notification is pretty simple on purpose:

https://github.com/readthedocs/addons/blob/46b2a96f7fba707130e61c8bfc0b03149411fc16/src/notification.js#L241-L250

The `nightly` slug is detected as a non-stable version and it shows that notification. The only slug we treat as latest version is `latest` 😄 . You may want to disable this notification for now until we think more about it. We could add other names there maybe like `development`, `nightly`, `main`, `dev`, etc... or just let the user to define what version should be treat as "latest".

**Comment by ktbarrett** [comment]
@stsewd I'm not sure if this is the greatest place to request this, but can this feature also be enabled for `cocotb`?
